signalfx/resource_signalfx_data_link.go

```diff
@@ -161,11 +161,19 @@ func getPayloadDataLink(d *schema.ResourceData) (*datalink.CreateUpdateDataLinkR
 		}
 	}
 
+	if name, ok := d.GetOk("property_name"); ok {
+		dataLink.PropertyName = name.(string)
+	}
+
 	if val, ok := d.GetOk("context_dashboard_id"); ok {
 		dataLink.ContextId = val.(string)
 	}
 
 	if val, ok := d.GetOk("target_signalfx_dashboard"); ok {
+		if name, ok := d.GetOk("property_name"); !ok || name.(string) == "" {
```

Comment:
Do we need to strip name before doing a whitespace comparison? Or is checking for == "" not even needed?

Reply:
Wait, is it a thing that property name is required for links to dashboards, but not the other types?!

Reply:
Digging into the code I see that GetOk already checks if it's a zero value. So I guess the == "" isn't needed unless extra whitespace is a concern. But maybe TF handles this higher up in the stack so that whitespace only gets normalized to empty string?

Reply:
Yep. I discovered this in the UI. If you click through the different data link types you'll see the trigger options are different for the different types. Only dashboard targets require a property name best I can tell.

Reply:
Fascinating!

Ok, will look over this with that in mind.

Reply:
Ok, I would just look at `dataLink.PropertyName` in the above code, since you've already verified it there. Also, we should probably make a DOCS ticket that asks to clarify this since I don't think this is explained in the API docs.